### PR TITLE
Example under chain decomposition.

### PR DESCRIPTION
**Pull request by Lukong123**
https://github.com/networkx/networkx/issues/5484

### CODE REVIEW

File: networkx/algorithms/chains.py [hunk 1]
```diff
@@ -49,6 +49,11 @@ def chain_decomposition(G, root=None):
     The worst-case running time of this implementation is linear in the
     number of nodes and number of edges [1]_.
 
+    Examples
+    --------
+    >>> G = nx.path_graph(5)
+    >>> chain_d = nx.chain_decomposition(G)
+
```
Comment (rossbar) [inline networkx/algorithms/chains.py:56]: Thanks for adding the example @Lukong123 , but I think we'll need to work on it a bit more to make it really useful for docstring readers!

In it's current state, the example shows how to *call* the function, but doesn't really give the user any sense of how the function behaves. The first challenge here is to come up with a good example graph - you want to choose one that has properties which are relevant for the function in the example (e.g. in this case, the example graph should probably have a chain).

Typically the best way to give users a sense of what a function does is to show it's output. In this case, `chain_decomposition` actually creates a [generator](https://wiki.python.org/moin/Generators) which isn't really that informative until it is iterated over! For example, one easy way to consume the objects from a generator is to create a list from it, e.g. `list(nx.chain_decomposition(G))`. Note how the result is empty (this goes back to the first point of choosing a good example graph)!

Anyways these are just a few things to keep in mind when coming up with good docstring examples!

Comment (Lukong123) [reply networkx/algorithms/chains.py:56]: Thank you, sir, I will keep that in mind.

**Comment by Lukong123** [comment]
Please sir @rossbar  check out this new example, I hope it is a much better one compared to the previous one.

**Comment by rossbar** [comment]
I think the merge conflicts here arise from the fact that this also incorporates the changes from the recent `edge_cover` examples, likely due to accidentally branching from the edge cover branch.

There are multiple ways that this could be fixed e.g. via rebasing. I'd be happy to help with this, though it would involve force-pushing and would require you to reset your local branch. If you're not comfortable with that, another option would be to create a new branch (from `main`) and resubmit the relevant `chain_decomposition` example in a new PR. LMK if you'd like me to rebase for you, otherwise feel free to fix in whatever way you're most comfortable with!

**Comment by rossbar** [comment]
Oops - it turns out this is related to the changes introduced to `covering.py` in #5549, not changes to the examples. Anyways the above still stands - the edge covering examples would be better in a separate PR so that this one focuses only on chain_decomposition.

**Comment by Lukong123** [comment]
> Oops - it turns out this is related to the changes introduced to `covering.py` in #5549, not changes to the examples. Anyways the above still stands - the edge covering examples would be better in a separate PR so that this one focuses only on chain_decomposition.

**Comment by Lukong123** [comment]
> I think the merge conflicts here arise from the fact that this also incorporates the changes from the recent `edge_cover` examples, likely due to accidentally branching from the edge cover branch.
> 
> There are multiple ways that this could be fixed e.g. via rebasing. I'd be happy to help with this, though it would involve force-pushing and would require you to reset your local branch. If you're not comfortable with that, another option would be to create a new branch (from `main`) and resubmit the relevant `chain_decomposition` example in a new PR. LMK if you'd like me to rebase for you, otherwise feel free to fix in whatever way you're most comfortable with!

Okay sir, thanks a lot. It looks like I had conflicts in some pull requests i.e #5485, #5534, #5502, #5540, and #5485 and most of them are conflicting with  `covering.py` so sir, I guess a better and easier way for me will be to create a separate pull request for the different examples.

**Comment by MridulS** [comment]
this was added in https://github.com/networkx/networkx/pull/5641, closing this PR. Thanks @Lukong123 !

**Pull request closed without merging**